Main.elm

```diff
@@ -84,24 +108,8 @@ view address model =
     [ div
         []
         (model
-          |> Debug.log "model"
           |> List.indexedMap (,)
-          |> List.map
-              (\( index, tuple ) ->
-                div
-                  []
-                  [ input
-                      [ value (fst tuple)
-                      , on "input" targetValue (\str -> Signal.message address (UpdateKey ( index, str )))
-                      ]
-                      []
-                  , input
-                      [ value (snd tuple)
-                      , on "input" targetValue (\str -> Signal.message address (UpdateValue ( index, str )))
-                      ]
-                      []
-                  ]
-              )
+          |> List.map (createInput address)
```

Comment:
Now `createInput` is pulled out I can partially apply it with the `address` and use that to map over the results.